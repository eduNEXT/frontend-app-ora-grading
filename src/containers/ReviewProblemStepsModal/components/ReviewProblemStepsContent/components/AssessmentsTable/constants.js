export const receivedAssessmentData = [
  {
    idAssessment: '55550453040',
    reviewerName: 'Carlos Doe',
    userName: 'carlos_doe_10',
    email: 'carlos@email.com',
    assessmentDate: '28-10-2023',
    assessmentScores: [{
      id: 'aB3cD7eF',
      type: 'Ideas',
      quality: 'Fair',
      rate: 1,
    },
    {
      id: '9GhI2jKl',
      type: 'Content',
      quality: 'Excellent',
      rate: 5,
    }],
    problemStep: 'Staff',
    feedback: 'This is working really well',
  },
  {
    idAssessment: '55550453042',
    reviewerName: 'John Smith',
    userName: 'john_smith_23',
    email: 'john@email.com',
    assessmentDate: '30-10-2023',
    assessmentScores: [{
      id: 'M4nO8pQr',
      type: 'Ideas',
      quality: 'Good',
      rate: 5,
    },
    {
      id: 'X5tU6vWx',
      type: 'Content',
      quality: 'Excellent',
      rate: 5,
    }],
    problemStep: 'Training',
    feedback: 'Great progress!',
  },
  {
    idAssessment: '55550453048',
    reviewerName: 'Emily Johnson',
    userName: 'emily_j',
    email: 'emily@email.com',
    assessmentDate: '29-10-2023',
    assessmentScores: [{
      id: 'PqW0uVwX',
      type: 'Ideas',
      quality: null,
      rate: null,
    },
    {
      id: 'ZabC12DE',
      type: 'Content',
      quality: null,
      rate: null,
    }],
    problemStep: 'Peers',
    feedback: 'Needs improvement in certain areas.',
  },
  {
    idAssessment: '55550453050',
    reviewerName: 'Sarah Brown',
    userName: 'sarah_brown_45',
    email: 'sarah@email.com',
    assessmentDate: '31-10-2023',
    assessmentScores: [{
      id: 'Yz123456',
      type: 'Ideas',
      quality: 'Good',
      rate: 3,
    },
    {
      id: 'L7iR9oSt',
      type: 'Content',
      quality: 'Excellent',
      rate: 5,
    }],
    problemStep: 'Self',
    feedback: 'Excellent work, keep it up!',
  },
];

export const givenAssessmentData = receivedAssessmentData.map(assessment => ({
  ...assessment,
  learnerName: assessment.reviewerName,
}));
