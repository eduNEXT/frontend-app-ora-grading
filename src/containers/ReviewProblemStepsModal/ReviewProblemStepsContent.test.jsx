import React from 'react';
import { shallow } from '@edx/react-unit-test-utils';

import { selectors } from 'data/redux';
import { RequestKeys } from 'data/constants/requests';
import {
  ReviewProblemStepsContent,
  mapStateToProps,
} from './ReviewProblemStepsContent';

jest.mock('data/redux', () => ({
  selectors: {
    app: {
      showRubric: (...args) => ({ showRubric: args }),
    },
    requests: {
      isCompleted: (...args) => ({ isCompleted: args }),
      isFailed: (...args) => ({ isFailed: args }),
    },
  },
}));
jest.mock('containers/ResponseDisplay', () => 'ResponseDisplay');
jest.mock('containers/Rubric', () => 'Rubric');
jest.mock('containers/ReviewModal/ReviewErrors', () => 'ReviewErrors');

describe('ReviewContent component', () => {
  describe('component', () => {
    describe('render tests', () => {
      test('snapshot: not loaded, no error', () => {
        expect(shallow(<ReviewProblemStepsContent />).isEmptyRender()).toEqual(true);
      });
      test('snapshot: show rubric', () => {
        expect(shallow(<ReviewProblemStepsContent isLoaded />)).toMatchSnapshot();
      });
      test('snapshot: hide rubric', () => {
        expect(shallow(<ReviewProblemStepsContent isLoaded showRubric />)).toMatchSnapshot();
      });
      test('snapshot: failed, showRubric (errors only)', () => {
        expect(shallow(<ReviewProblemStepsContent showRubric isFailed />)).toMatchSnapshot();
      });
    });
  });
  describe('mapStateToProps', () => {
    let mapped;
    const testState = { some: 'test-state' };
    beforeEach(() => {
      mapped = mapStateToProps(testState);
    });
    const requestKey = RequestKeys.fetchSubmission;
    test('showRubric loads from app.showRubric', () => {
      expect(mapped.showRubric).toEqual(selectors.app.showRubric(testState));
    });
    test('isFailed loads from requests.isFailed(fetchSubmission)', () => {
      expect(mapped.isFailed).toEqual(selectors.requests.isFailed(testState, { requestKey }));
    });
    test('isLoadeed loads from requests.isCompleted(fetchSubmission)', () => {
      expect(mapped.isLoaded).toEqual(selectors.requests.isCompleted(testState, { requestKey }));
    });
  });
});
