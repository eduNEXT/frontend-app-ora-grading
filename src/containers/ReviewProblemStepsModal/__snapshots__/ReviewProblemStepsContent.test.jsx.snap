// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewContent component component render tests snapshot: failed, showRubric (errors only) 1`] = `
ReactShallowRenderer {
  "shallowRenderer": ReactShallowRenderer {
    "_context": Object {},
    "_didScheduleRenderPhaseUpdate": false,
    "_dispatcher": Object {
      "readContext": [Function],
      "useCallback": [Function],
      "useContext": [Function],
      "useDebugValue": [Function],
      "useDeferredValue": [Function],
      "useEffect": [Function],
      "useId": [Function],
      "useImperativeHandle": [Function],
      "useInsertionEffect": [Function],
      "useLayoutEffect": [Function],
      "useMemo": [Function],
      "useReducer": [Function],
      "useRef": [Function],
      "useResponder": [Function],
      "useState": [Function],
      "useSyncExternalStore": [Function],
      "useTransition": [Function],
    },
    "_element": <ReviewProblemStepsContent
      isFailed={true}
      isLoaded={false}
      showRubric={true}
    />,
    "_firstWorkInProgressHook": null,
    "_forcedUpdate": false,
    "_idCounter": 0,
    "_instance": null,
    "_isReRender": false,
    "_newState": null,
    "_numberOfReRenders": 0,
    "_renderPhaseUpdates": null,
    "_rendered": <div
      className="content-block"
    >
      <div
        className="content-wrapper"
      >
        <ReviewErrors />
      </div>
    </div>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "_workInProgressHook": null,
  },
  "shallowWrapper": <div
    className="content-block"
  >
    <div
      className="content-wrapper"
    >
      <ReviewErrors />
    </div>
  </div>,
}
`;

exports[`ReviewContent component component render tests snapshot: hide rubric 1`] = `
ReactShallowRenderer {
  "shallowRenderer": ReactShallowRenderer {
    "_context": Object {},
    "_didScheduleRenderPhaseUpdate": false,
    "_dispatcher": Object {
      "readContext": [Function],
      "useCallback": [Function],
      "useContext": [Function],
      "useDebugValue": [Function],
      "useDeferredValue": [Function],
      "useEffect": [Function],
      "useId": [Function],
      "useImperativeHandle": [Function],
      "useInsertionEffect": [Function],
      "useLayoutEffect": [Function],
      "useMemo": [Function],
      "useReducer": [Function],
      "useRef": [Function],
      "useResponder": [Function],
      "useState": [Function],
      "useSyncExternalStore": [Function],
      "useTransition": [Function],
    },
    "_element": <ReviewProblemStepsContent
      isFailed={false}
      isLoaded={true}
      showRubric={true}
    />,
    "_firstWorkInProgressHook": null,
    "_forcedUpdate": false,
    "_idCounter": 0,
    "_instance": null,
    "_isReRender": false,
    "_newState": null,
    "_numberOfReRenders": 0,
    "_renderPhaseUpdates": null,
    "_rendered": <div
      className="content-block"
    >
      <div
        className="content-wrapper"
      >
        <ReviewErrors />
        <Row
          className="flex-nowrap m-0"
        >
          <Col
            className="p-0"
          >
            <ResponseDisplay />
          </Col>
          <Rubric />
        </Row>
      </div>
    </div>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "_workInProgressHook": null,
  },
  "shallowWrapper": <div
    className="content-block"
  >
    <div
      className="content-wrapper"
    >
      <ReviewErrors />
      <Row
        className="flex-nowrap m-0"
      >
        <Col
          className="p-0"
        >
          <ResponseDisplay />
        </Col>
        <Rubric />
      </Row>
    </div>
  </div>,
}
`;

exports[`ReviewContent component component render tests snapshot: show rubric 1`] = `
ReactShallowRenderer {
  "shallowRenderer": ReactShallowRenderer {
    "_context": Object {},
    "_didScheduleRenderPhaseUpdate": false,
    "_dispatcher": Object {
      "readContext": [Function],
      "useCallback": [Function],
      "useContext": [Function],
      "useDebugValue": [Function],
      "useDeferredValue": [Function],
      "useEffect": [Function],
      "useId": [Function],
      "useImperativeHandle": [Function],
      "useInsertionEffect": [Function],
      "useLayoutEffect": [Function],
      "useMemo": [Function],
      "useReducer": [Function],
      "useRef": [Function],
      "useResponder": [Function],
      "useState": [Function],
      "useSyncExternalStore": [Function],
      "useTransition": [Function],
    },
    "_element": <ReviewProblemStepsContent
      isFailed={false}
      isLoaded={true}
      showRubric={false}
    />,
    "_firstWorkInProgressHook": null,
    "_forcedUpdate": false,
    "_idCounter": 0,
    "_instance": null,
    "_isReRender": false,
    "_newState": null,
    "_numberOfReRenders": 0,
    "_renderPhaseUpdates": null,
    "_rendered": <div
      className="content-block"
    >
      <div
        className="content-wrapper"
      >
        <ReviewErrors />
        <Row
          className="flex-nowrap m-0"
        >
          <Col
            className="p-0"
          >
            <ResponseDisplay />
          </Col>
        </Row>
      </div>
    </div>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "_workInProgressHook": null,
  },
  "shallowWrapper": <div
    className="content-block"
  >
    <div
      className="content-wrapper"
    >
      <ReviewErrors />
      <Row
        className="flex-nowrap m-0"
      >
        <Col
          className="p-0"
        >
          <ResponseDisplay />
        </Col>
      </Row>
    </div>
  </div>,
}
`;
